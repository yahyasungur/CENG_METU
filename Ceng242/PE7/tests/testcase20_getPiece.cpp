#include "submain.h"

int main() {

	Piece* pieces[256];
	for (int i=0; i<256; i++) {
		StraightEdge edge1(i*4);
		StraightEdge edge2(i*4 + 1);
		StraightEdge edge3(i*4 + 2);
		StraightEdge edge4(i*4 + 3);
		Edge* edges[4] = {&edge1, &edge2, &edge3, &edge4};

		pieces[i] = new Piece(edges);
	}

	Puzzle* puzzle = new Puzzle(16, 0, 0);
	for (int i=0; i<16; i++) {
		for (int j=0; j<16; j++) {
			// place piece
			int location[2] = {i, j};
			puzzle->placePiece(*pieces[j*16 + i], location);

			// get piece
			try {
				int location[2] = {j, i};
				const Piece& piece = puzzle->getPiece(location);
				cout << "The following piece has been found:" << endl;
				cout << piece << endl;
			}
			catch(exception& e) {
				cout << e.what() << endl << endl;
			}
		}
	}

	cout << *puzzle << endl;

	for (int i=0; i<256; i++)
		delete pieces[i];
	delete puzzle;

	return 0;
}