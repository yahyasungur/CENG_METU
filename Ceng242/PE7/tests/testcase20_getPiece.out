The following piece has been found:
0  
1  
2  
3  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
64  
65  
66  
67  

The following piece has been found:
68  
69  
70  
71  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
128  
129  
130  
131  

The following piece has been found:
132  
133  
134  
135  

The following piece has been found:
136  
137  
138  
139  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
192  
193  
194  
195  

The following piece has been found:
196  
197  
198  
199  

The following piece has been found:
200  
201  
202  
203  

The following piece has been found:
204  
205  
206  
207  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
256  
257  
258  
259  

The following piece has been found:
260  
261  
262  
263  

The following piece has been found:
264  
265  
266  
267  

The following piece has been found:
268  
269  
270  
271  

The following piece has been found:
272  
273  
274  
275  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
320  
321  
322  
323  

The following piece has been found:
324  
325  
326  
327  

The following piece has been found:
328  
329  
330  
331  

The following piece has been found:
332  
333  
334  
335  

The following piece has been found:
336  
337  
338  
339  

The following piece has been found:
340  
341  
342  
343  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
384  
385  
386  
387  

The following piece has been found:
388  
389  
390  
391  

The following piece has been found:
392  
393  
394  
395  

The following piece has been found:
396  
397  
398  
399  

The following piece has been found:
400  
401  
402  
403  

The following piece has been found:
404  
405  
406  
407  

The following piece has been found:
408  
409  
410  
411  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
448  
449  
450  
451  

The following piece has been found:
452  
453  
454  
455  

The following piece has been found:
456  
457  
458  
459  

The following piece has been found:
460  
461  
462  
463  

The following piece has been found:
464  
465  
466  
467  

The following piece has been found:
468  
469  
470  
471  

The following piece has been found:
472  
473  
474  
475  

The following piece has been found:
476  
477  
478  
479  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
512  
513  
514  
515  

The following piece has been found:
516  
517  
518  
519  

The following piece has been found:
520  
521  
522  
523  

The following piece has been found:
524  
525  
526  
527  

The following piece has been found:
528  
529  
530  
531  

The following piece has been found:
532  
533  
534  
535  

The following piece has been found:
536  
537  
538  
539  

The following piece has been found:
540  
541  
542  
543  

The following piece has been found:
544  
545  
546  
547  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
576  
577  
578  
579  

The following piece has been found:
580  
581  
582  
583  

The following piece has been found:
584  
585  
586  
587  

The following piece has been found:
588  
589  
590  
591  

The following piece has been found:
592  
593  
594  
595  

The following piece has been found:
596  
597  
598  
599  

The following piece has been found:
600  
601  
602  
603  

The following piece has been found:
604  
605  
606  
607  

The following piece has been found:
608  
609  
610  
611  

The following piece has been found:
612  
613  
614  
615  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
640  
641  
642  
643  

The following piece has been found:
644  
645  
646  
647  

The following piece has been found:
648  
649  
650  
651  

The following piece has been found:
652  
653  
654  
655  

The following piece has been found:
656  
657  
658  
659  

The following piece has been found:
660  
661  
662  
663  

The following piece has been found:
664  
665  
666  
667  

The following piece has been found:
668  
669  
670  
671  

The following piece has been found:
672  
673  
674  
675  

The following piece has been found:
676  
677  
678  
679  

The following piece has been found:
680  
681  
682  
683  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
704  
705  
706  
707  

The following piece has been found:
708  
709  
710  
711  

The following piece has been found:
712  
713  
714  
715  

The following piece has been found:
716  
717  
718  
719  

The following piece has been found:
720  
721  
722  
723  

The following piece has been found:
724  
725  
726  
727  

The following piece has been found:
728  
729  
730  
731  

The following piece has been found:
732  
733  
734  
735  

The following piece has been found:
736  
737  
738  
739  

The following piece has been found:
740  
741  
742  
743  

The following piece has been found:
744  
745  
746  
747  

The following piece has been found:
748  
749  
750  
751  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
768  
769  
770  
771  

The following piece has been found:
772  
773  
774  
775  

The following piece has been found:
776  
777  
778  
779  

The following piece has been found:
780  
781  
782  
783  

The following piece has been found:
784  
785  
786  
787  

The following piece has been found:
788  
789  
790  
791  

The following piece has been found:
792  
793  
794  
795  

The following piece has been found:
796  
797  
798  
799  

The following piece has been found:
800  
801  
802  
803  

The following piece has been found:
804  
805  
806  
807  

The following piece has been found:
808  
809  
810  
811  

The following piece has been found:
812  
813  
814  
815  

The following piece has been found:
816  
817  
818  
819  

The cell that you request is empty!

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
832  
833  
834  
835  

The following piece has been found:
836  
837  
838  
839  

The following piece has been found:
840  
841  
842  
843  

The following piece has been found:
844  
845  
846  
847  

The following piece has been found:
848  
849  
850  
851  

The following piece has been found:
852  
853  
854  
855  

The following piece has been found:
856  
857  
858  
859  

The following piece has been found:
860  
861  
862  
863  

The following piece has been found:
864  
865  
866  
867  

The following piece has been found:
868  
869  
870  
871  

The following piece has been found:
872  
873  
874  
875  

The following piece has been found:
876  
877  
878  
879  

The following piece has been found:
880  
881  
882  
883  

The following piece has been found:
884  
885  
886  
887  

The cell that you request is empty!

The cell that you request is empty!

The following piece has been found:
896  
897  
898  
899  

The following piece has been found:
900  
901  
902  
903  

The following piece has been found:
904  
905  
906  
907  

The following piece has been found:
908  
909  
910  
911  

The following piece has been found:
912  
913  
914  
915  

The following piece has been found:
916  
917  
918  
919  

The following piece has been found:
920  
921  
922  
923  

The following piece has been found:
924  
925  
926  
927  

The following piece has been found:
928  
929  
930  
931  

The following piece has been found:
932  
933  
934  
935  

The following piece has been found:
936  
937  
938  
939  

The following piece has been found:
940  
941  
942  
943  

The following piece has been found:
944  
945  
946  
947  

The following piece has been found:
948  
949  
950  
951  

The following piece has been found:
952  
953  
954  
955  

The cell that you request is empty!

The following piece has been found:
960  
961  
962  
963  

The following piece has been found:
964  
965  
966  
967  

The following piece has been found:
968  
969  
970  
971  

The following piece has been found:
972  
973  
974  
975  

The following piece has been found:
976  
977  
978  
979  

The following piece has been found:
980  
981  
982  
983  

The following piece has been found:
984  
985  
986  
987  

The following piece has been found:
988  
989  
990  
991  

The following piece has been found:
992  
993  
994  
995  

The following piece has been found:
996  
997  
998  
999  

The following piece has been found:
1000  
1001  
1002  
1003  

The following piece has been found:
1004  
1005  
1006  
1007  

The following piece has been found:
1008  
1009  
1010  
1011  

The following piece has been found:
1012  
1013  
1014  
1015  

The following piece has been found:
1016  
1017  
1018  
1019  

The following piece has been found:
1020  
1021  
1022  
1023  

0  64  128  192  256  320  384  448  512  576  640  704  768  832  896  960  
1  65  129  193  257  321  385  449  513  577  641  705  769  833  897  961  
2  66  130  194  258  322  386  450  514  578  642  706  770  834  898  962  
3  67  131  195  259  323  387  451  515  579  643  707  771  835  899  963  

4  68  132  196  260  324  388  452  516  580  644  708  772  836  900  964  
5  69  133  197  261  325  389  453  517  581  645  709  773  837  901  965  
6  70  134  198  262  326  390  454  518  582  646  710  774  838  902  966  
7  71  135  199  263  327  391  455  519  583  647  711  775  839  903  967  

8  72  136  200  264  328  392  456  520  584  648  712  776  840  904  968  
9  73  137  201  265  329  393  457  521  585  649  713  777  841  905  969  
10  74  138  202  266  330  394  458  522  586  650  714  778  842  906  970  
11  75  139  203  267  331  395  459  523  587  651  715  779  843  907  971  

12  76  140  204  268  332  396  460  524  588  652  716  780  844  908  972  
13  77  141  205  269  333  397  461  525  589  653  717  781  845  909  973  
14  78  142  206  270  334  398  462  526  590  654  718  782  846  910  974  
15  79  143  207  271  335  399  463  527  591  655  719  783  847  911  975  

16  80  144  208  272  336  400  464  528  592  656  720  784  848  912  976  
17  81  145  209  273  337  401  465  529  593  657  721  785  849  913  977  
18  82  146  210  274  338  402  466  530  594  658  722  786  850  914  978  
19  83  147  211  275  339  403  467  531  595  659  723  787  851  915  979  

20  84  148  212  276  340  404  468  532  596  660  724  788  852  916  980  
21  85  149  213  277  341  405  469  533  597  661  725  789  853  917  981  
22  86  150  214  278  342  406  470  534  598  662  726  790  854  918  982  
23  87  151  215  279  343  407  471  535  599  663  727  791  855  919  983  

24  88  152  216  280  344  408  472  536  600  664  728  792  856  920  984  
25  89  153  217  281  345  409  473  537  601  665  729  793  857  921  985  
26  90  154  218  282  346  410  474  538  602  666  730  794  858  922  986  
27  91  155  219  283  347  411  475  539  603  667  731  795  859  923  987  

28  92  156  220  284  348  412  476  540  604  668  732  796  860  924  988  
29  93  157  221  285  349  413  477  541  605  669  733  797  861  925  989  
30  94  158  222  286  350  414  478  542  606  670  734  798  862  926  990  
31  95  159  223  287  351  415  479  543  607  671  735  799  863  927  991  

32  96  160  224  288  352  416  480  544  608  672  736  800  864  928  992  
33  97  161  225  289  353  417  481  545  609  673  737  801  865  929  993  
34  98  162  226  290  354  418  482  546  610  674  738  802  866  930  994  
35  99  163  227  291  355  419  483  547  611  675  739  803  867  931  995  

36  100  164  228  292  356  420  484  548  612  676  740  804  868  932  996  
37  101  165  229  293  357  421  485  549  613  677  741  805  869  933  997  
38  102  166  230  294  358  422  486  550  614  678  742  806  870  934  998  
39  103  167  231  295  359  423  487  551  615  679  743  807  871  935  999  

40  104  168  232  296  360  424  488  552  616  680  744  808  872  936  1000  
41  105  169  233  297  361  425  489  553  617  681  745  809  873  937  1001  
42  106  170  234  298  362  426  490  554  618  682  746  810  874  938  1002  
43  107  171  235  299  363  427  491  555  619  683  747  811  875  939  1003  

44  108  172  236  300  364  428  492  556  620  684  748  812  876  940  1004  
45  109  173  237  301  365  429  493  557  621  685  749  813  877  941  1005  
46  110  174  238  302  366  430  494  558  622  686  750  814  878  942  1006  
47  111  175  239  303  367  431  495  559  623  687  751  815  879  943  1007  

48  112  176  240  304  368  432  496  560  624  688  752  816  880  944  1008  
49  113  177  241  305  369  433  497  561  625  689  753  817  881  945  1009  
50  114  178  242  306  370  434  498  562  626  690  754  818  882  946  1010  
51  115  179  243  307  371  435  499  563  627  691  755  819  883  947  1011  

52  116  180  244  308  372  436  500  564  628  692  756  820  884  948  1012  
53  117  181  245  309  373  437  501  565  629  693  757  821  885  949  1013  
54  118  182  246  310  374  438  502  566  630  694  758  822  886  950  1014  
55  119  183  247  311  375  439  503  567  631  695  759  823  887  951  1015  

56  120  184  248  312  376  440  504  568  632  696  760  824  888  952  1016  
57  121  185  249  313  377  441  505  569  633  697  761  825  889  953  1017  
58  122  186  250  314  378  442  506  570  634  698  762  826  890  954  1018  
59  123  187  251  315  379  443  507  571  635  699  763  827  891  955  1019  

60  124  188  252  316  380  444  508  572  636  700  764  828  892  956  1020  
61  125  189  253  317  381  445  509  573  637  701  765  829  893  957  1021  
62  126  190  254  318  382  446  510  574  638  702  766  830  894  958  1022  
63  127  191  255  319  383  447  511  575  639  703  767  831  895  959  1023  


